frontends/concrete-python/concrete/fhe/mlir/processors/assign_bit_widths.py

```diff
@@ -88,13 +88,38 @@ def max_encrypted_bitwidth_node(node: Node):
         return normal_p + 1
 
     if name in MULTIPLY and all(value.is_encrypted for value in node.inputs):
-        return normal_p + 1
+        # For operations that use multiply, an additional bit
+        # needs to be added to the bitwidths of the inputs.
+        # For single precision circuits the max of the input / output
+        # precisions will be taken in required_encrypted_bitwidth. For
```

Comment:
I think this comment should not be there. And it's not how it work for now.
First partitions with block of homegenuous precision based on operation that can/cannot change precisions are delimited. All operations in a block gets the same output/input precisions. Then for operation that can change precision like TLU or TLU variants it accepts to have a different input precision to account for transition from other blocks of precisions.